recipes/libdisplay-info/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import cross_building
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import copy, get, replace_in_file, rm, rmdir
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.layout import basic_layout
+from conan.tools.meson import Meson, MesonToolchain
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class LibdisplayInfoConan(ConanFile):
+    name = "libdisplay-info"
+    description = "EDID and DisplayID library."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://gitlab.freedesktop.org/emersion/libdisplay-info"
+    topics = ("display", "DisplayID", "EDID")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _has_build_profile(self):
+        return getattr(self, "settings_build", None)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        if not self._has_build_profile:
+            self.requires("hwdata/0.374")
+
+    def validate(self):
+        if not self.settings.os in ["FreeBSD", "Linux"]:
+            raise ConanInvalidConfiguration(f"{self.ref} is not supported on {self.settings.os}")
+
+    def build_requirements(self):
+        if self._has_build_profile:
+            self.tool_requires("hwdata/0.374")
```

Comment:
```suggestion
            self.tool_requires("hwdata/<host_version>")
```

https://docs.conan.io/2/reference/conanfile/methods/build_requirements.html#host-version

Reply:
I don't think that will work in this case since `hwdata` is only a required in the `host` context for single-profile builds, otherwise it is only a `tool_requires`.